e2e/cypress/integration/messaging/receive_message_on_socket_reconnect_spec.js

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+import users from '../../fixtures/users.json';
+
+let townsquareChannelId;
+const websockets = [];
+
+describe('Messaging', () => {
+    before(() => {
+        // # Wrap websocket to be able to connect and close connections on demand
+        cy.mockWebsockets(websockets);
+
+        // # Login and go to /
+        cy.apiLogin('sysadmin');
+        cy.visit('/');
+
+        // # Get ChannelID to use later
+        cy.getCurrentChannelId().then((id) => {
+            townsquareChannelId = id;
+        });
+    });
+
+    it('M18682-RHS fetches messages on socket reconnect when a different channel is in center', () => {
+        // # Connect all sockets
+        websockets.forEach((value) => {
+            value.connect();
+        });
+
+        // # Post a message as another user
+        cy.postMessageAs({sender: users['user-1'], message: 'abc', channelId: townsquareChannelId}).wait(TIMEOUTS.SMALL);
+
+        // # Click "Reply"
+        cy.getLastPostId().then((rootPostId) => {
+            cy.clickPostCommentIcon(rootPostId);
+
+            // # Post a message
+            cy.postMessageReplyInRHS('def');
+
+            // # Save our post for later use
+            cy.getLastPostIdRHS().as('myPostId');
+
+            // # Change channel
+            cy.get('#sidebarItem_suscipit-4').click({force: true}).then(() => {
+                // # Close all sockets
+                websockets.forEach((value) => {
+                    if (value.close) {
+                        value.close();
+                    }
+                });
+
+                // # Post message as a different user
+                cy.postMessageAs({sender: users['user-1'], message: 'ghi', channelId: townsquareChannelId, rootId: rootPostId});
+
+                // # Wait a short time to check whether the message appears or not
+                cy.wait(TIMEOUTS.SMALL);
+
+                // * Verify that only "def" is posted and not "ghi"
+                cy.get('#rhsPostList').should('be.visible').children().should('have.length', 1);
+                cy.get('#rhsPostList').within(() => {
+                    cy.findByText('def').should('be.visible');
+                    cy.queryByText('ghi').should('not.exist');
```

Comment:
Could you check once more that the test is passing on your side.  This is failing on me.  Looks like the websocket is not being closed and `ghi` gets posted immediately.

Reply:
The test is passing on my side. Can you check whether you are seeing the stub messages on the messages list on cypress?
![image](https://user-images.githubusercontent.com/1933730/71166645-7774e180-2253-11ea-8ff8-8318c540d26f.png)

Reply:
My bad. It is not working. I will fix it on the next push.